src/components/Admin/Authentication/ChangeName.tsx

```diff
@@ -0,0 +1,223 @@
+  Text,
+  Button,
+  Box,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  FormErrorMessage,
+  FormHelperText,
+} from '@chakra-ui/react';
+import {CheckCircleIcon} from '@chakra-ui/icons';
+import {
+  handleReauthenticationWithPassword,
+  PasswordFormInput,
+  SubmitButton,
+} from './Util';
+import {firestore, firebaseAuth} from '../../../firebase';
+import {useTranslation} from 'react-i18next';
+
+/**
+ * Props for ChangeNameModal component. Used for type safety.
+ */
+interface ChangeNameModalProps {
+  passwordUser: boolean;
+}
+
+/**
+ * Component for changing an authenticated user's name. Includes button that
+ * opens modal to update (or set for the first time) the user's name.
+ * @param props - passwordUser
+ * - `passwordUser`: if user uses a password for authentication, used to
+ *                   reauthenticate if necessary before updating the user's name
+ */
+const ChangeNameModal: ({
+  passwordUser,
+}: ChangeNameModalProps) => JSX.Element = ({
+  passwordUser,
+}: ChangeNameModalProps) => {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+
+  // Current password state variables
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [passwordVisible, setPasswordVisible] = useState(false);
+
+  const [newName, setNewName] = useState('');
+  const [error, setError] = useState('');
+  const [modalIsLoading, setModalIsLoading] = useState(false);
+  const [nameChangeComplete, setNameChangeComplete] = useState(false);
+  // --------------- End state maintenance variables ------------------------
+
+  const {t} = useTranslation(['administration', 'common']);
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose() {
+    setPassword('');
+    setPasswordError('');
+    setPasswordVisible(false);
+    setNewName('');
+    setError('');
+    setModalIsLoading(false);
+    setNameChangeComplete(false);
+    onClose();
+  }
+
+  /**
+   * Updates an authenticated user using Firebase authentication
+   * @param event - submit form event
+   */
+  function handleDisplayNameUpdate(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    setModalIsLoading(true);
+
+    /**
+     * Updates the user's name in Firebase and in their Firestore user doc
+     */
+    function updateName(): void {
+      if (firebaseAuth.currentUser) {
+        const user = firebaseAuth.currentUser;
+        // Update Firebase account
+        user
+          .updateProfile({
+            displayName: newName,
+          })
+          .then(() => {
+            // Update Firestore user document
+            firestore
+              .collection('users')
+              .doc(user.uid)
+              .update({
+                name: newName,
+              })
+              .then(() => setNameChangeComplete(true))
+              .catch(error => {
```

Comment:
Does firebase require us to catch and rethrow vs. never catching to begin with?

Reply:
Good point, I can just remove the catch statement ([this Stack Overflow](https://stackoverflow.com/questions/42167274/rethrowing-error-in-promise-catch) post explains it well)